File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.empty()) return "";
        string base = strs[0];
        for(int i = 0; i < base.size(); i++){
            string prefix = base.substr(0, i + 1);
            for(int j = 1; j < strs.size(); j++){
                if(strs[j].find(prefix) != 0){
                    return base.substr(0, i);
                }
            }
        } 
        // Longest Common Prefix
        return base;
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length == 0) return "";
        
        String base = strs[0];
        for(int i = 0; i < base.length(); i++){
            String r = base.substring(0, i + 1);
            for(int j = 1; j < strs.length; j++){
                if(!strs[j].startsWith(r)){
                    return base.substring(0, i);
                }
            }
        }
        return base;
    }
}